pallets/orml-tokens-extension/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+#![deny(warnings)]
+#![cfg_attr(test, feature(proc_macro_hygiene))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+extern crate mocktopus;
+
+#[cfg(test)]
+use mocktopus::macros::mockable;
+use orml_traits::MultiCurrency;
+use sp_std::{convert::TryInto, prelude::*, vec};
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[cfg(test)]
+mod mock;
+
+pub mod default_weights;
+
+#[cfg(test)]
+mod tests;
+
+mod ext;
+
+mod types;
+
+pub use pallet::*;
+
+pub(crate) type BalanceOf<T> = <<T as orml_currencies::Config>::MultiCurrency as MultiCurrency<
+	<T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub(crate) type CurrencyOf<T> = <<T as orml_currencies::Config>::MultiCurrency as MultiCurrency<
+	<T as frame_system::Config>::AccountId,
+>>::CurrencyId;
+
+pub(crate) type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use crate::types::CurrencyDetails;
+	use frame_support::{pallet_prelude::*, transactional};
+	use frame_system::{ensure_root, ensure_signed, pallet_prelude::OriginFor};
+
+	pub use default_weights::WeightInfo;
+
+	/// ## Configuration
+	/// The pallet's configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + orml_tokens::Config + orml_currencies::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Weight information for the extrinsics in this module.
+		type WeightInfo: WeightInfo;
+
+		/// Type that allows for checking if currency type is ownable by users
+		type CurrencyIdChecker: CurrencyIdCheck<CurrencyId = CurrencyOf<Self>>;
+
+		/// TODO needs to be conditionaly compiled
+		#[cfg(feature = "runtime-benchmarks")]
+		type GetTestCurrency: Get<CurrencyOf<Self>>;
```

Comment:
We don't necessarily need this config type at all. Instead of passing a currency just for testing and benchmarking purposes, we can either (in tests and benchmarks):
- get the native currency ID from teh `orml_currencies` pallet, like [here](https://github.com/pendulum-chain/pendulum/blob/10f3b7fa66995a0cfa93b26511209db47481e3c6/pallets/orml-currencies-allowance-extension/src/benchmarking.rs#L13)
- or just define the `CurrencyId` type as a simple `u64` and use numbers in the tests and benchmarks, see [here](https://github.com/pendulum-chain/pendulum/blob/10f3b7fa66995a0cfa93b26511209db47481e3c6/pallets/orml-currencies-allowance-extension/src/mock.rs#L36) and [here](https://github.com/pendulum-chain/pendulum/blob/10f3b7fa66995a0cfa93b26511209db47481e3c6/pallets/orml-currencies-allowance-extension/src/benchmarking.rs#L17)

And then we can also adjust the `CurrencyIdChecker` to always allow `u64` numbers in the mock.

Reply:
I actually had some troubles when using the `concrete` currency in the benchmarks. 
- Native currency I tried but throws an error when minting.
- Second solution would be fine, I tried to use the non-generic `CurrencyId:Token(u64)` on the benchmarks but couldn't make that work due to type mismatch, that's why I added the conditional config type. 
I didn't try with u64, I'll give it a try.